### PR TITLE
Object store/ aws testing

### PR DESCRIPTION
**Pull request by deepsidhu85**
## Description of changes
What did you change in this pull request?  Provide a description of files changed, user interactions changed, etc.  Include how to test your changes.

- Added testing for AWS S3 object storage. All methods in the `IridaFileStorageAWSUtilityImpl.java` are tested against a dockerized mock of `S3` storage provided by Adobe
- Updated `com.amazonaws:aws-java-sdk-s3` maven package to the latest version
- Updated the `checkWriteAccess` method in `IridaFileStorageAWSUtilityImpl.java` to attempt to read/write from/to the `S3` bucket if the bucket permissions are not available

To run the tests locally you need to run the command `docker run -p 9090:9090 -p 9191:9191 -t adobe/s3mock` in a terminal. This will pull the `s3mock` docker image and start up the service.

## Related issue
Link to the GitHub issue this pull request addresses using the `#issuenum` format.  If it completes an issue, use `Fixes #issuenum` to automatically close the issue.

## Checklist
Things for the developer to confirm they've done before the PR should be accepted:

~* [ ] CHANGELOG.md (and UPGRADING.md if necessary) updated with information for new change.~
* [X] Tests added (or description of how to test) for any new features.
~* [ ] User documentation updated for UI or technical changes.~

### CODE REVIEW

File: run-tests.sh [hunk 1]
```diff
@@ -106,9 +107,15 @@ test_service() {
 }
 
 test_file_system() {
+  docker run -d -p 9090:9090 -p 9191:9191 --name $S3MOCK_DOCKER_NAME -t adobe/s3mock
+    ./gradlew clean check fileSystemITest -Dspring.datasource.url=$JDBC_URL -Dfile.processing.decompress=true -Dirida.it.rootdirectory=$TMP_DIRECTORY -Dspring.datasource.dbcp2.max-wait=$DB_MAX_WAIT_MILLIS $@
```
Comment (ericenns) [inline run-tests.sh:111]: Please remove this duplicate execution of the tests. line 111 and 112

**Review by ericenns — CHANGES_REQUESTED** [review]
Looks good, except filesystem integration tests are being run twice. Failing first and then succeeding.

File: run-tests.sh (hunk 1 above)
Comment (deepsidhu85) [reply run-tests.sh]: Removed in 52e42e91096b76089f29d43565068647ca7dc85c. Good catch :)

**Review by ericenns — APPROVED** [review]
👍 Everything looks good now, thanks for adding this testing

**Pull request merged**